### ISSUE TITLE
Screen Reader does not read keyboard dialog

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/friction#165]
For https://github.com/phetsims/QA/issues/244.  Found with NVDA on Firefox.  There does not seem to be anything for the screen reader to read when opening the keyboard nav dialog.  It only reads "blank."

**Comment by KatieWoe** [comment]
This is not the case with Jaws + chrome

**Comment by KatieWoe** [comment]
Edit: I managed to get it to read the dialog by activating keyboard nav and having it focus on the x.

**Comment by KatieWoe** [comment]
Also found in https://github.com/phetsims/QA/issues/247

**Comment by zepumph** [comment]
I found that on chrome, there is only one thing to tab to in the sim when the dialog is open: the close button. On Firefox there are two focusable items. There is the close button, but before it (in focus order) there is something else that doesn't have any focus highlight. When you focus it it says "blank" when logging the `document.activeElement` when it was focused, it printed the whole `html` element. When focusing that element, the screen reader read "blank" and the virtual cursor didn't navigate into the dialog, but only repeated blank. 

I'm not sure why this is happening only here, and not in chrome. It may have to do with the complete removal of all other content, rather than using more of the aria spec for modal dialogs https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html.

So @KatieWoe to answer the root of this issue, the screen reader doesn't  read the dialog because it isn't actually "in" the dialog, but when you focus the close button, you enter the dialog and can use the virtual cursor. 

The two concrete strangeties for investigation seem to be:
1. Why does the dialog in firefox add the `html` element to the focus order?
2. When the `html` element is focused, why can't you use the virtual cursor to navigate into the dialog.

**Comment by jessegreenberg** [comment]
I found something similar happens in the published version of BASE but instead of getting stuck on "blank", NVDA gets stuck on "Keyboard Shortcuts".

**Comment by jessegreenberg** [comment]
Even without NVDA I am finding that the HTML element is focusable sometimes.

**Comment by jessegreenberg** [comment]
I was able to produce the problem with this basic HTML as well:
<details>

```html
<html>
<body>
  <p>Virtual cursor hits here always, but can't enter this dialog from the outside</p>
 <div role="dialog" aria-labelledby="dialog1Title" aria-describedby="dialog1Desc">
 <h2 id="dialog1Title">Subscription Form</h2>
 <form>
   <p>
     <label for="firstName">First Name</label>
     <input id="firstName" type="text" />
   </p>
   <p>
       <input type="submit" value="Save Information"/>
   </p>
 </form>
</div>
</body>
</html>
```
</details>

The NVDA virtual cursor cannot enter an element with `role=Dialog` from the outside. When I use the virtual cursor, it says "Subscription Form" and gets stuck there. When I press "Enter", the virtual cursor enters the Dialog. This is also true for PhET dialogs.

So the remaining issue is that the NVDA says "Blank" instead of the dialog title. We removed the `aria-describedby` that probably would fix this, Ill try to find why.

**Comment by jessegreenberg** [comment]
We removed `aria-describedby` in https://github.com/phetsims/balloons-and-static-electricity/issues/399 because the dialog sounds better in VO without it. We decided that VO was too verbose with aria-describedby.

**Comment by jessegreenberg** [comment]
AboutDialog has
```js
    // a11y - set label association so the title is read when focus enters the dialog
    this.addAriaLabelledbyAssociation( {
      thisElementName: AccessiblePeer.PRIMARY_SIBLING,
      otherNode: titleText,
      otherElementName: AccessiblePeer.PRIMARY_SIBLING
    } );
```
I think we should include this for all Dialogs that use options.title. If not, have something similar in KeyboardHelpDialog. This way, Dialog label is read when this happens and user doesn't hear "blank". @zepumph does this sound reasonable?

**Comment by zepumph** [comment]
I found that we are in fact already doing this in dialog.js, see 

```js

    // a11y - set the aria-labelledby relation so that whenever focus enters the dialog the title is read
    if ( options.title && options.title.tagName ) {
      this.addAriaLabelledbyAssociation( {
        thisElementName: AccessiblePeer.PRIMARY_SIBLING,
        otherNode: options.title,
        otherElementName: AccessiblePeer.PRIMARY_SIBLING
      } );
    }

```

The problem is that the `options.title` node is actually a VBox, set here:
```js
    // stack the two items with a bit of spacing
    options.title = new VBox( {
        children: [
          shortcutsTitleText,

          // labelWithIcon is meant to be passed to HelpContent, so we have to hack a bit here
          new HBox( { children: [ tabHintLine.icon, tabHintLine.label ], spacing: 4 } )
        ],
        spacing: 5
      }
    );
```

so I'm going to add an accessibleName to that vbox too

**Comment by jbphet** [comment]
@zepumph - Is there any update on this, or an idea of when it might be addressed?  After today, it is likely to be the only issue blocking publication of RIAW v1.6.

**Comment by zepumph** [comment]
Yes thank you @jbphet, the above commit should fix this, please add it to your branch, as I will for Friction.

**Comment by KatieWoe** [comment]
@jbphet @zepumph I just tested in https://github.com/phetsims/QA/issues/257.  It still will not read the entire dialog if keyboard focus is not on the x.  However, it does read "Keyboard shortcuts, tab to get started."  Is this acceptable?

**Comment by jessegreenberg** [comment]
That is the correct behavior for NVDA @KatieWoe, thanks for checking. We found that to be the behavior for NVDA for all dialogs, even outside of PhET, so this is what we want (tested in https://github.com/phetsims/friction/issues/165#issuecomment-452349367).

**Comment by zepumph** [comment]
From here I will cherry-pick this into Friction 1.5

**Comment by zepumph** [comment]
Commit has been cherry-picked into branch (note to self)

**Comment by KatieWoe** [comment]
1.5.0-rc.2 looks good on NVDA
